# Project: [Exploring Airbnb Market Trends](https://app.datacamp.com/learn/projects/exploring-airbnb-market-trends)

![NYC Skyline](nyc.jpg)

Welcome to New York City, one of the most-visited cities in the world. There are many Airbnb listings in New York City to meet the high demand for temporary lodging for travelers, which can be anywhere between a few nights to many months. In this project, we will take a closer look at the New York Airbnb market by combining data from multiple file types like `.csv`, `.tsv`, and `.xlsx`.

Recall that **CSV**, **TSV**, and **Excel** files are three common formats for storing data. 
Three files containing data on 2019 Airbnb listings are available to you:

**data/airbnb_price.csv**
This is a CSV file containing data on Airbnb listing prices and locations.
- **`listing_id`**: unique identifier of listing
- **`price`**: nightly listing price in USD
- **`nbhood_full`**: name of borough and neighborhood where listing is located

**data/airbnb_room_type.xlsx**
This is an Excel file containing data on Airbnb listing descriptions and room types.
- **`listing_id`**: unique identifier of listing
- **`description`**: listing description
- **`room_type`**: Airbnb has three types of rooms: shared rooms, private rooms, and entire homes/apartments

**data/airbnb_last_review.tsv**
This is a TSV file containing data on Airbnb host names and review dates.
- **`listing_id`**: unique identifier of listing
- **`host_name`**: name of listing host
- **`last_review`**: date when the listing was last reviewed

Original project version info:
- Pandas version: 1.5.1

In [14]:
# import our packages
import pandas as pd
print(f'Pandas version: {pd.__version__}')

Pandas version: 2.1.4


##  Load in our data, inspect and clean

***
### Load in the airbnb_price.csv data

In [15]:
price_df = pd.read_csv('data/airbnb_price.csv')
display(price_df.head())
print(price_df.info())

listing_id        price                nbhood_full
0        2595  225 dollars         Manhattan, Midtown
1        3831   89 dollars     Brooklyn, Clinton Hill
2        5099  200 dollars     Manhattan, Murray Hill
3        5178   79 dollars  Manhattan, Hell's Kitchen
4        5238  150 dollars       Manhattan, Chinatown

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 25209 entries, 0 to 25208
Data columns (total 3 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   listing_id   25209 non-null  int64 
 1   price        25209 non-null  object
 2   nbhood_full  25209 non-null  object
dtypes: int64(1), object(2)
memory usage: 591.0+ KB
None


**Brief summary of the price dataframe**
- Has 25209 records
- Has no _NULL_ values. 
- Will need to remove the `' dollars'` text from the `price` field and convert the column to float. 



In [16]:
price_df['price'] = price_df['price'].str.replace(' dollars', '').astype('float')
display(price_df.head())
print(price_df.price.dtype)

listing_id  price                nbhood_full
0        2595  225.0         Manhattan, Midtown
1        3831   89.0     Brooklyn, Clinton Hill
2        5099  200.0     Manhattan, Murray Hill
3        5178   79.0  Manhattan, Hell's Kitchen
4        5238  150.0       Manhattan, Chinatown

float64


***
### Load in the airbnb_room_type.xlsx data

In [17]:
room_type_df = pd.read_excel('data/airbnb_room_type.xlsx')
display(room_type_df.head())
print(room_type_df.info())


listing_id                                description        room_type
0        2595                      Skylit Midtown Castle  Entire home/apt
1        3831            Cozy Entire Floor of Brownstone  Entire home/apt
2        5099  Large Cozy 1 BR Apartment In Midtown East  Entire home/apt
3        5178            Large Furnished Room Near B'way     private room
4        5238         Cute & Cozy Lower East Side 1 bdrm  Entire home/apt

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 25209 entries, 0 to 25208
Data columns (total 3 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   listing_id   25209 non-null  int64 
 1   description  25199 non-null  object
 2   room_type    25209 non-null  object
dtypes: int64(1), object(2)
memory usage: 591.0+ KB
None


In [18]:
# check the values in the room_type column
room_type_df.room_type.value_counts()

room_type
Entire home/apt    8458
Private room       7241
entire home/apt    2665
private room       2248
ENTIRE HOME/APT    2143
PRIVATE ROOM       1867
Shared room         380
shared room         110
SHARED ROOM          97
Name: count, dtype: int64

In [19]:
# check for nulls in our columns
room_type_df.isnull().sum()

listing_id      0
description    10
room_type       0
dtype: int64

**Brief summary of the room_type dataframe**
- Has 25209 records
- Has 10 _NULL_ values in the `description` column
- The `room_type` column has values which are the same but have different casing so will need to be corrected.


In [20]:
# set all the room_type values to lower case
room_type_df['room_type'] = room_type_df.room_type.str.lower()
room_type_df.room_type.value_counts()

room_type
entire home/apt    13266
private room       11356
shared room          587
Name: count, dtype: int64

***
### Load in our airbnb_last_review.tsv data

In [21]:
last_review_df = pd.read_csv('data/airbnb_last_review.tsv', sep='\t', parse_dates=['last_review'])
display(last_review_df.head())
print(last_review_df.info())

listing_id    host_name last_review
0        2595     Jennifer  2019-05-21
1        3831  LisaRoxanne  2019-07-05
2        5099        Chris  2019-06-22
3        5178     Shunichi  2019-06-24
4        5238          Ben  2019-06-09

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 25209 entries, 0 to 25208
Data columns (total 3 columns):
 #   Column       Non-Null Count  Dtype         
---  ------       --------------  -----         
 0   listing_id   25209 non-null  int64         
 1   host_name    25201 non-null  object        
 2   last_review  25209 non-null  datetime64[ns]
dtypes: datetime64[ns](1), int64(1), object(1)
memory usage: 591.0+ KB
None


In [22]:
# print out the number of nulls for each column
last_review_df.isnull().sum()

listing_id     0
host_name      8
last_review    0
dtype: int64

**Brief summary of the last_review dataframe**
- Has 25209 records
- Has 8 _NULL_ values in the `host_name` column


***
## Answer the questions
1. What are the dates of the earliest and most recent reviews? Store these values as two separate variables with your preferred names.
2. How many of the listings are private rooms? Save this into any variable.
3. What is the average listing price? Round to the nearest penny and save into a variable.


### 1. What are the dates of the earliest and most recent reviews?

In [23]:
earliest_review = last_review_df.last_review.min()
most_recent_review = last_review_df.last_review.max()
print(earliest_review)
print(most_recent_review)

2019-01-01 00:00:00
2019-07-09 00:00:00


### 2. How many of the listings are private rooms?

In [24]:
num_private_rooms = room_type_df.query('room_type == "private room"')['room_type'].count()
print(num_private_rooms)

11356


### 3. What is the average listing price? 

In [25]:
average_price = price_df.price.mean().round(2)
print(average_price)

141.78


Combine the new variables into one DataFrame called `review_dates` with four columns in the following order: `first_reviewed`, `last_reviewed`, `nb_private_rooms`, and `avg_price`. The DataFrame should only contain one row of values.

In [26]:
review_dates = pd.DataFrame(
    {'first_reviewed': [earliest_review],
     'last_reviewed': [most_recent_review],
     'nb_private_rooms': [num_private_rooms],
     'avg_price': [average_price]
    })
review_dates

first_reviewed last_reviewed  nb_private_rooms  avg_price
0     2019-01-01    2019-07-09             11356     141.78